src/sections/provider/components/ProviderSelectButton/ProviderSelectButton.tsx

```diff
@@ -0,0 +1,46 @@
+import { useBestNumber } from "api/chain"
+import { PROVIDERS, useProviderRpcUrlStore } from "api/provider"
+import { Separator } from "components/Separator/Separator"
+import { Text } from "components/Typography/Text/Text"
+import { useState } from "react"
+import { ReactComponent as ChevronRightIcon } from "assets/icons/ChevronRightIcon.svg"
+import { ProviderSelectModal } from "sections/provider/ProviderSelectModal"
+import { ProviderStatus } from "sections/provider/ProviderStatus"
+import { SButton, SName } from "./ProviderSelectButton.styled"
+
+export const ProviderSelectButton = () => {
+  const [open, setOpen] = useState(false)
+  const store = useProviderRpcUrlStore()
+
+  const rpcUrl = store.rpcUrl ?? import.meta.env.VITE_PROVIDER_URL
+  const selectedProvider = PROVIDERS.find((provider) => provider.url === rpcUrl)
+  const number = useBestNumber()
+
+  return (
+    <>
+      <SButton tabIndex={0} onClick={() => setOpen(true)} whileHover="animate">
+        <SName
+          variants={{
+            initial: { width: 0 },
+            animate: { width: "auto" },
+            exit: { width: 0 },
```

Comment:
myslim ze exit animace ti nebudou bezet bez `AnimatePresence`

Reply:
funguje to